hack/functests.sh

```diff
@@ -37,4 +35,43 @@ if [[ ${KUBEVIRT_PROVIDER} == os-* ]] || [[ ${KUBEVIRT_PROVIDER} =~ (okd|ocp)-*
     oc=${kubectl}
 fi
 
-${TESTS_OUT_DIR}/tests.test -kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -test.timeout 420m ${FUNC_TEST_ARGS} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra}
+rm -rf $ARTIFACTS
+mkdir -p $ARTIFACTS
+
+function functest() {
+    _out/tests/ginkgo -r $@ _out/tests/tests.test -- -kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra} -config=${KUBEVIRT_DIR}/tests/default-config.json --artifacts=${ARTIFACTS}
```

Comment:
This looks weird somehow. Why is it required to pass the test binary now? Does the ginkgo test runner itself take care of the parallel execution somehow?

Reply:
Directly executing the test binaries allows running the tests in non-parallel fashion only. The ginkgo binary contains the actual parallel test runner. This runner starts the test binary n-times and distributes via streams the specs to run to the binaries. I think it makes sense to also call the non-parallel execution now via the ginkgo binary to have a common way of invoking the tests.

Reply:
Thanks for the explanation!